crates/specs/src/state.rs

```diff
@@ -1,23 +1,44 @@
+use serde::Deserialize;
 use serde::Serialize;
 
-#[derive(Clone, Debug, Serialize)]
-pub struct InitializationState<T> {
-    pub eid: T,
-    pub fid: T,
-    pub iid: T,
-    pub frame_id: T,
-    pub sp: T,
-
-    pub host_public_inputs: T,
-    pub context_in_index: T,
-    pub context_out_index: T,
-    pub external_host_call_call_index: T,
-
-    pub initial_memory_pages: T,
-    pub maximal_memory_pages: T,
-
-    #[cfg(feature = "continuation")]
-    pub jops: T,
+cfg_if::cfg_if! {
+    if #[cfg(feature = "continuation")] {
+        #[derive(Clone, Debug, Serialize, Deserialize)]
+        pub struct InitializationState<T> {
+            pub eid: T,
+            pub fid: T,
+            pub iid: T,
+            pub frame_id: T,
+            pub sp: T,
+
+            pub host_public_inputs: T,
+            pub context_in_index: T,
+            pub context_out_index: T,
+            pub external_host_call_call_index: T,
+
+            pub initial_memory_pages: T,
+            pub maximal_memory_pages: T,
+
+            pub jops: T,
```

Comment:
> #[cfg(feature = "continuation") ] for jops ?

Because  `Deserialize` trait doesn't work when using directly mark `#[cfg(feature = "continuation") ]` on ` jops` , so I use `cfg_if::cfg_if! `.